packages/xo-common/src/api-errors.js

```diff
@@ -200,3 +189,13 @@ export const notEnoughResources = create(24, data => ({
   data, // [{ resourceSet, resourceType, available, requested }]
   message: 'not enough resources in resource set',
 }))
+
+export const incorrectState = create(25, ({ actual, context, expected, ids }) => ({
+  data: {
+    actual,
+    context,
+    expected,
+    ids,
+  },
+  message: 'incorrect state',
+}))
```

Comment:
What about?
```js
throw incorrectState({
  property: 'runningJobs',
  expected: [],
  actual: runningJobIds
})
```

Or you could send simply the number of jobs, but I agree your proposal provide more data.

Reply:
`runningJobs` is not a property. why expected is empty? If this error throws it means that we expect something ?

Reply:
Maybe `object` would be better than `property`, and if necessary we'll be able to add this entry if necessary.

Reply:
I agree, `runningJobs` should be in `object`, not `property` :slightly_smiling_face: 

`expected` is not empty, it's the expected value, like: `assert(runningJobs, [])`.